src/common.c

```diff
@@ -537,26 +537,31 @@ int fi_epoll_del(struct fi_epoll *ep, int fd)
 	return -FI_EINVAL;
 }
 
-void *fi_epoll_wait(struct fi_epoll *ep, int timeout)
+int fi_epoll_wait(struct fi_epoll *ep, void **contexts, int max_contexts,
+                  int timeout)
 {
 	int i, ret;
+	int found = 0;
 
 	ret = poll(ep->fds, ep->nfds, timeout);
-	if (ret <= 0)
-		return NULL;
+	if (ret == -1)
```

Comment:
Just a one minor comment regarding this (it was missed in previous version too, but wasn't reproducible, because 
exactly `SOCKET_ERROR` < 0):
`WSAPoll` (the Windows' poll implementation) returns `SOCKET_ERROR`. I'm not sure that it's always -1. Could you please dadd the following stuf in the _include/unix/osd.h_:
```
#ifndef SOCKET_ERROR
#define SOCKET_ERROR (-1)
#endif
```
and use `SOCKET_ERROR` to check the return value from `poll` function

Thank you in advance